### PR TITLE
Add new backend for LinkedIn OpenID Connect

### PR DESCRIPTION
**Pull request by noxan**
## Proposed changes

LinkedIn OAuth2 has been deprecated as of August 2023 and been replace with OpenID connect.
This pull request adds LinkedIn OpenID Connect. It has a few specific adjustments as LinkedIn does not send any "nonce" and therefore does not work with the regular OpenID Connect implementation.

Thanks to #829 for the idea and the hint on the OpenID Connect auth base class.

## Types of changes

Please check the type of change your PR introduces:

- [ ] Release (new release request)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (PEP8, lint, formatting, renaming, etc)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes (build process, tests runner, etc)
- [ ] Other (please describe):

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask. We're here to
help! This is simply a reminder of what we are going to look for before merging
your code._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: social_core/backends/linkedin.py [hunk 1]
```diff
@@ -2,11 +2,57 @@
...
+
+        # nonce = id_token.get("nonce")
+        # if not nonce:
+        #     raise AuthTokenError(self, "Incorrect id_token: nonce")
+
+        # nonce_obj = self.get_nonce(nonce)
+        # if nonce_obj:
+        #     self.remove_nonce(nonce_obj.id)
+        # else:
+        #     raise AuthTokenError(self, "Incorrect id_token: nonce")
```
Comment (noxan) [inline social_core/backends/linkedin.py:53]: Maybe remove those comments? Just wanted to make it clear what had been disabled. Or we could add a flag to the base openid connect implementation to disable it there.

File: social_core/backends/linkedin.py [hunk 2]
```diff
@@ -2,11 +2,57 @@
...
+        if "nbf" in id_token and utc_timestamp < id_token["nbf"]:
+            raise AuthTokenError(self, "Incorrect id_token: nbf")
+
+        # Verify the token was issued in the last 10 minutes
+        iat_leeway = self.setting("ID_TOKEN_MAX_AGE", self.ID_TOKEN_MAX_AGE)
+        if utc_timestamp > id_token["iat"] + iat_leeway:
+            raise AuthTokenError(self, "Incorrect id_token: iat")
+
+        # Skip the nonce validation for linkedin as it does not provide any nonce.
+        # https://stackoverflow.com/questions/76889585/issues-with-sign-in-with-linkedin-using-openid-connect
```
Comment (noxan) [inline social_core/backends/linkedin.py:43]: I also verified this on my own setup, LinkedIn does indeed not provide a `nonce` in the payload.

File: social_core/backends/linkedin.py [hunk 3]
```diff
@@ -2,11 +2,57 @@
...
+class LinkedinOpenIdConnect(OpenIdConnectAuth):
+    """
+    Linkedin OpenID Connect backend. Oauth2 has been deprecated as of August 1, 2023.
+    https://learn.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/sign-in-with-linkedin-v2?context=linkedin/consumer/context
+    """
+
+    name = "linkedin-openidconnect"
+    # Settings from https://www.linkedin.com/oauth/.well-known/openid-configuration
+    OIDC_ENDPOINT = "https://www.linkedin.com/oauth"
+
```
Comment (noxan) [inline social_core/backends/linkedin.py:23]:
```suggestion
    DEFAULT_SCOPE = ["openid", "profile", "email"]

```

File: social_core/backends/linkedin.py (hunk 1 above)
Comment (nijel) [reply social_core/backends/linkedin.py:53]: Commend describing it is not needed is good enough, commented out code is not really needed.

Comment (noxan) [reply social_core/backends/linkedin.py:53]:
```suggestion
```

**Comment by noxan** [comment]
@nijel sorry I missed your initial comment, resolved your feedback. anything else keeping us from adding this pull request?

**Comment by nijel** [comment]
Can you please also add tests for this?

**Comment by nijel** [comment]
...and documentation to https://github.com/python-social-auth/social-docs

**Comment by noxan** [comment]
haha, sure. i'll look into it :)

**Comment by nijel** [comment]
Merged, thanks for your contribution! Please followup with the documentation PR.

**Pull request merged**